### PR TITLE
feat: Add a check to validate the presence of git

### LINKED ISSUE #84: Add a check to validate the presence of git
- The uno bootstrapper needs the presence of the CLI version of git, which may not be installed automatically by Visual Studio.
- Provide a suggestion to include it from the visual studio IDE if on Windows, or steps for macOS and Linux

### PR DESCRIPTION
**Pull request by aesteves900**
GitHub Issue (If applicable): closes #84   

# PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

People from the community have reported that certain projects fail to build because of long paths.

There is no check to Git Cli presence on Linux, macOS, and Windows.

## What is the new behavior?

The uno-check will validate the presence of git cli.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current Figma desktop app
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Unit Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes and the __PlugIn is BACKWARD COMPATIBLE__
- [x] Associated with an issue

### CODE REVIEW

File: UnoCheck/Checkups/GitCliInstalledCheckup.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				new GitCliSolution())));
+			}
+
+			if (Util.IsLinux)
+			{
+				var shellRunnerAboutLinuxResult = ShellProcessRunner.Run("lsb_release", "-a");
+
+				var linuxRelease = shellRunnerAboutLinuxResult.GetOutput().Trim();
+
+				if (shellHasGitRunnerResult.ExitCode != 0 && (linuxRelease.Contains(Ubuntu) || linuxRelease.Contains(Debian)))
```
Comment (jeromelaban) [inline UnoCheck/Checkups/GitCliInstalledCheckup.cs:38]:
```suggestion
				var isDebianBased = (linuxRelease.Contains(Ubuntu) || linuxRelease.Contains(Debian));
				if (shellHasGitRunnerResult.ExitCode != 0 && isDebianBased)
```

File: UnoCheck/Solutions/GitCliSolution.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			await base.Implement(sharedState, cancellationToken);
+
+			var ps = new ProcessStartInfo(WinCliUrl)
+			{
+				UseShellExecute = true,
+				Verb = "open"
+			};
+			_ = Process.Start(ps);
+
+			ReportStatus("The Phyton version can be downloaded from Windows Store.");
```
Comment (jeromelaban) [inline UnoCheck/Solutions/GitCliSolution.cs:22]: This needs to be updated. If running on Windows, you can also suggest to use the visual studio installer.

File: UnoCheck/Solutions/LinuxGitCliSolution.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	internal class LinuxGitCliSolution : Solution
+	{
+		public override async Task Implement(SharedState sharedState, CancellationToken cancellationToken)
+		{
+			await base.Implement(sharedState, cancellationToken);
+			
+			var r = await Util.WrapShellCommandWithSudo("apt-get", new[] { "install", "git" });
+			ReportStatus(r.GetOutput());
+
+			//ReportStatus("Git CLI was installed on Linux.");
```
Comment (jeromelaban) [inline UnoCheck/Solutions/LinuxGitCliSolution.cs:16]: Should be removed?

File: UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+			{
+				UseShellExecute = true,
+				Verb = "open"
+			};
+			_ = Process.Start(ps);
+
+			ReportStatus("For other Linux distributions, please check the Git Cli web page.");
+		}
+
+		private const string PythonDownloadUrl = "https://git-scm.com/book/en/v2/Getting-Started-Installing-Git";
```
Comment (jeromelaban) [inline UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs:24]: Update the variable name

File: UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		public override async Task Implement(SharedState sharedState, CancellationToken cancellationToken)
+		{
+			await base.Implement(sharedState, cancellationToken);
+
+			var ps = new ProcessStartInfo(PythonDownloadUrl)
+			{
+				UseShellExecute = true,
+				Verb = "open"
+			};
+			_ = Process.Start(ps);
```
Comment (jeromelaban) [inline UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs:19]: Can you validate that this works on a Linux environment? I'm not sure if the verb is supported.

Comment (aesteves900) [reply UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs]: It needed to fix on Fedora. Done!

File: UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs (hunk 4 above)
Comment (aesteves900) [reply UnoCheck/Solutions/LinuxOtherDistGitCliSolution.cs]: Done!

File: UnoCheck/Solutions/GitCliSolution.cs (hunk 2 above)
Comment (aesteves900) [reply UnoCheck/Solutions/GitCliSolution.cs]: You were right! Done!

File: UnoCheck/Checkups/GitCliInstalledCheckup.cs (hunk 1 above)
Comment (aesteves900) [reply UnoCheck/Checkups/GitCliInstalledCheckup.cs]: Done!

File: UnoCheck/Solutions/LinuxGitCliSolution.cs (hunk 3 above)
Comment (aesteves900) [reply UnoCheck/Solutions/LinuxGitCliSolution.cs]: Done!

File: UnoCheck/Solutions/GitCliSolution.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+						UseShellExecute = true,
+						Verb = "open"
+					};
+					_ = Process.Start(vs);
+
+					ReportStatus("Please, install via Visual Studio Installer");
+				}
+				else
+				{
+					ReportStatus("Visual Studio Installer was not found, ");
```
Comment (jeromelaban) [inline UnoCheck/Solutions/GitCliSolution.cs:37]:
```suggestion
					ReportStatus("Visual Studio Installer was not found.");
```

File: UnoCheck/Checkups/GitCliInstalledCheckup.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+			//just for macOS: If you don’t have it installed already, it will prompt you to install it.
+			var shellHasGitRunnerResult = ShellProcessRunner.Run("git", "--version");
+
+			if (Util.IsWindows && shellHasGitRunnerResult.ExitCode != 0)
+			{
+				return await Task.FromResult(new DiagnosticResult(
+				Status.Error,
+				this,
+				new Suggestion(InstallMessage,
+				new GitCliSolution())));
```
Comment (jeromelaban) [inline UnoCheck/Checkups/GitCliInstalledCheckup.cs:29]:
```suggestion
				return await Task.FromResult(
					new DiagnosticResult(
						Status.Error,
						this,
						new Suggestion(InstallMessage, new GitCliSolution())));
```

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**